src/components/oscal-utils/OSCALLinkUtils.js

```diff
@@ -58,3 +58,76 @@ export function guessExtensionFromHref(url) {
   }
   return detectedExtension;
 }
+
+/**
+ * Finds if a fragment contains a control group by searching the control tabs
+ *
+ * @param {string} fragment A given fragment
+ * @returns {string} The control group id
+ */
+export function determineControlGroupFromFragment(fragment) {
+  if (fragment === null || fragment === "" || fragment === undefined) {
+    return null;
+  }
+  // Create array from all tab control grouping elements
+  const controlGroupList = Array.from(
+    document.querySelectorAll('[id^="vertical-tab-"]')
+  );
+  // Grab group id if found
+  return controlGroupList
+    ?.find(
+      (group) =>
+        fragment.split("/")?.[0] === group.id.split("vertical-tab-").pop()
+    )
+    ?.id?.split("vertical-tab-")
+    ?.pop();
+}
+
+/**
+ * Add (push) a control/sub-control to fragmentPrefix.
+ *
+ * @param {string} fragmentPrefix The beginning of a fragment
+ * @param {string} controlId The identification for a control
+ * @returns {string} fragmentPrefix with an added control
+ */
+export function appendToFragmentPrefix(fragmentPrefix, controlId) {
+  return fragmentPrefix && fragmentPrefix !== ""
+    ? `${fragmentPrefix}/${controlId}`
+    : controlId;
+}
+
+/**
+ * Remove (shift) a grouping from a fragmentSuffix.
+ *
+ * @param {string} fragmentSuffix The end of a fragment
+ * @returns {string} fragmentSuffix with a removed control
+ */
+export function shiftFragmentSuffix(fragmentSuffix) {
+  return fragmentSuffix && fragmentSuffix !== ""
+    ? `${fragmentSuffix.substring(fragmentSuffix.indexOf("/") + 1)}`
+    : null;
+}
+
+/**
+ * Transforms text to prepare as a fragment, by replacing slashes and spaces with hyphens and
+ * lowercasing all letters.
+ *
+ * @param {string} linkText Text to format
+ * @returns {string} Formatted text to work as a fragment
+ */
+export function conformLinkIdText(linkText) {
+  return !linkText
+    ? ""
+    : linkText?.props?.children.replace(/\\| |\//g, "-")?.toLowerCase() ||
+        linkText.replace(/\\| |\//g, "-")?.toLowerCase();
+}
+
+/**
+ * Validate a fragment.
+ *
+ * @param {string} fragment A given fragment
+ * @returns {boolean} State of it being a valid fragment
+ */
+export function isValidFragment(fragment) {
+  return fragment && fragment !== "";
+}
```

Comment:
This seems to just be checking that a fragment isn't falsy right? I think this function can either be made more generic, or removed for a simple boolean check.

Reply:
Yep -- I think `!!fragment` would accomplish the same goal if we need boolean coercion.